twine/utils.py

```diff
@@ -237,11 +238,31 @@ def get_userpass_value(
     if cli_value is not None:
         logger.info(f"{key} set by command options")
         return cli_value
+
     elif config.get(key) is not None:
         logger.info(f"{key} set from config file")
         return config[key]
+
     elif prompt_strategy:
-        return prompt_strategy()
+        warning = ""
+        value = prompt_strategy()
+
+        if not value:
+            warning = f"Your {key} is empty"
+        elif any(unicodedata.category(c).startswith("C") for c in value):
+            # See https://www.unicode.org/reports/tr44/#General_Category_Values
+            # Most common case is "\x16" when pasting in Windows Command Prompt
+            warning = f"Your {key} contains control characters"
+
+        if warning:
+            logger.warning(f"  {warning}. Did you enter it correctly?")
+            # TODO: Link to new entry in Twine docs
+            logger.warning(
+                "  See https://pypi.org/help/#invalid-auth for more information."
+            )
+
+        return value
```

Comment:
@sigmavirus24 What do you think of something like this? I haven't tried it on Windows (yet), but here's what it looks like with an empty password:

```
Uploading distributions to https://test.pypi.org/legacy/
Enter your username: brian
Enter your password: 
  Your password is empty. Did you enter it correctly?
  See https://pypi.org/help/#invalid-auth for more information.
Uploading twine-3.4.2.dev12+g6694f57.d20210619-py3-none-any.whl
100%|██████████| 41.7k/41.7k [00:00<00:00, 153kB/s] 
For more details, retry the upload with the --verbose option.
HTTPError: 403 Forbidden from https://test.pypi.org/legacy/
Invalid or non-existent authentication information. See https://test.pypi.org/help/#invalid-auth for more information.
```

If this seems reasonable, I'm happy to add a new section to Twine's docs, to keep this independent of a specific repository.

Reply:
LGTM